### PR TITLE
Implement #435 - Rename validation report

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR provides support to rename the validation report as the `feed_info.feed_publisher_name` + `__` + timestamp + file extension (`.json` or `.pb`). All whitespace characters are replaced by `_` before using the resulting path.

**Expected behavior:** 

**1. Validation report as `.json`:**
Validation reports should no longer be named `results.json`. Instead the name of said file should be `feed_info.feed_publisher_name` + `__` `timestamp`(at the time of validation in `America/Montreal` timezone) +`.json`.

If `feed_info.txt` is not provided then the name of the validation report is limited to the timestamp at the moment of validation in `America/Montreal` timezone.

**Sample filename:** after validating [MBTA's GTFS archive](https://cdn.mbta.com/MBTA_GTFS.zip) on 2020-10-20 at 09:07:48 (America/Montreal timezone), the validation report will be named as follows `MBTA__2020-10-20_09/07/48.442365.json` 

**2. Validation report as `.pb`:** 
Validation reports should no longer be named  `1.pb` /  `2.pb`/ ... / `n.pb` . Instead the same of said file should be `feed_info.feed_publisher_name` + `__` + `timestamp` (at the time of validation in `America/Montreal` timezone) + `-` + fileCounter + `.pb`.

If `feed_info.txt` is not provided then the name of the validation reports are limited to: `__` + `timestamp` (at the moment of validation in `America/Montreal` timezone) + filecounter + `.pb`

**Sample filenames:** after validating [MBTA's GTFS archive](https://cdn.mbta.com/MBTA_GTFS.zip) on 2020-10-20 at 09:07:48 (America/Montreal timezone), the validation reports will be named as follows 
- `MBTA__2020-10-20 09/07/48.442365-1.pb` 
- `MBTA__2020-10-20 09/07/48.442365-2.pb` 
- ...
- `MBTA__2020-10-20 09/07/48.442365-n.pb` 

image

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: adapter/exporter/src/main/java/org/mobilitydata/gtfsvalidator/exporter/ProtobufNoticeExporter.java [hunk 1]
```diff
@@ -1013,7 +1013,7 @@ public ProtobufOutputStreamGenerator(final String outputPath) {
 
         public OutputStream getStream() throws IOException {
             OutputStream newStream = Files.newOutputStream(Paths.get(
-                    targetPath + File.separator + streamCounter +
+                    targetPath + "-" + streamCounter +
```
Comment (fabrice-v) [inline adapter/exporter/src/main/java/org/mobilitydata/gtfsvalidator/exporter/ProtobufNoticeExporter.java:1016]: Just out of curiosity: what is this a fix for and is there any reason we'd run into troubles using the `-` (like it's an invalid character on some file systems or something)

**Review by fabrice-v — CHANGES_REQUESTED** [review]
Code looks good to me. 

2 changes required
- provide a sample filename in the PR description
- update the README to provide a sample filename

File: adapter/exporter/src/main/java/org/mobilitydata/gtfsvalidator/exporter/ProtobufNoticeExporter.java (hunk 1 above)
Comment (lionel-nj) [reply adapter/exporter/src/main/java/org/mobilitydata/gtfsvalidator/exporter/ProtobufNoticeExporter.java:1016]: Using `File.separator` caused an issue: the system could not find the path to `targetPath/streamcounter.pb/`.

**Review by fabrice-v — APPROVED** [review]
Good thks. As discussed, I'll see if I can reformat the string to exclude ` ` characters, they may come from `feed_info.feed_publisher_name` in addition to the timestamp.

I will also do the merge.

**Comment by fabrice-v** [comment]
This could be hardened by modifying the replace all regex to also replace those forbidden characters
`{ '\\', '/', ':', '*', '?', '"', '<', '>', '|', '[', ']', '\'', ';',
            '=', ',' }`

**Pull request merged**